pkg/systemd/overview-cards/shutdownStatus.jsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React, { useState, useEffect } from 'react';
+import { Button, Flex, FlexItem } from '@patternfly/react-core';
+import { PowerOffIcon, RedoIcon } from "@patternfly/react-icons";
+
+import * as timeformat from "timeformat";
+
+import cockpit from "cockpit";
+
+const _ = cockpit.gettext;
+const systemd_file = "/run/systemd/shutdown/scheduled";
+
+const getScheduledShutdown = (setShutdownTime, setShutdownType) => {
+    const client = cockpit.dbus("org.freedesktop.login1");
+    return client.call("/org/freedesktop/login1", "org.freedesktop.DBus.Properties", "Get", ["org.freedesktop.login1.Manager", "ScheduledShutdown"], { type: "ss" }).then(reply => {
+        const result = reply[0];
```

Comment:
This can be done in a bit less ugly way with structured decomposition: `.then(([type_v, time_v]) => { ... }`

Reply:
Not sure how that can be done, reply is an array of which contains a one object. so we can't destrcuct it easily

Reply:
You can at least deconstruct the array, so so `.then(([result]) => {...}` instead of your `const`. You can also decompose objects, but indeed that's not too useful here.